### ISSUE TITLE
Fetch validation results as an API call

### ISSUE THREAD

**Issue by bobsilverberg** [mozilla/addons-code-manager#1627]
When https://github.com/mozilla/addons-server/pull/14717 lands, which should fix https://github.com/mozilla/addons-server/issues/14655, which currently breaks code manager, we will need to change the logic we use to fetch validation results. 

Currently we simply fetch a URL, bit now we need to make an API call, which will require some logic changes in code-manager code.

**Comment by ioanarusiczki** [comment]
@bobsilverberg I verified this on AMO dev with an [addon](https://reviewers.addons-dev.allizom.org/en-US/reviewers/review/12roboform-12password-manager) :
- the validation endpoint is present in the Network with it's corresponding JSON response. I will give an example: 
https://addons-dev.allizom.org/api/v5/reviewers/addon/619715/file/411837/validation/
- for a compare page the validation file of the new version is present. 
- the files I've checked on dev or stage load and I could view then using the file viewer.
